vars/runGradleCheck.groovy

```diff
@@ -0,0 +1,59 @@
+void call(Map args = [:]) {
+    def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+    def git_repo_url = args.gitRepoUrl ?: 'null'
+    def git_reference = args.gitReference ?: 'null'
+
+    println("Git Repo: ${git_repo_url}")
+    println("Git Reference: ${git_reference}")
+
+    if (git_repo_url.equals('null') || git_reference.equals('null')) {
+        println("No git repo url or git reference to checkout the commit, exit 1")
+        System.exit(1)
+    }
+    else {
+        withCredentials([
+            usernamePassword(credentialsId: "jenkins-gradle-check-s3-aws-credentials", usernameVariable: 'amazon_s3_access_key', passwordVariable: 'amazon_s3_secret_key'),
+            usernamePassword(credentialsId: "jenkins-gradle-check-s3-aws-resources", usernameVariable: 'amazon_s3_base_path', passwordVariable: 'amazon_s3_bucket'),
+            usernamePassword(credentialsId: "jenkins-github-bot-token", usernameVariable: 'GITHUB_BOT_USER', passwordVariable: 'GITHUB_BOT_TOKEN')]) {
```

Comment:
Where are these used?

Reply:
These are used for connecting to an existing s3 bucket and test repository-s3 plugin functionality.
More details you can sync up with @dreamer-89 on it.
He created those credentials and demand the addition.

Thanks.